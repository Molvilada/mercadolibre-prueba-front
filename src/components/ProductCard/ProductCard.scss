@import "/src/styles/colors";

.ProductCard {
  cursor: pointer;
  padding: 18px;
  h2 {
    margin-top: 0px;
  }
  h3,
  h2 {
    font-weight: normal;
    span {
      display: none;
    }
  }

  hr {
    height: 1px;
    border: none;
    background-color: $gray-secondary;
  }
  .productImg {
    width: 180px;
    height: 180px;
    img {
      border-radius: 4px;
      width: 100%;
      height: 100%;
      object-fit: contain;
    }
  }

  .address {
    color: $text-secondary;
  }
}
