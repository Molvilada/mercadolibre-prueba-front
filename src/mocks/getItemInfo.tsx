export const getItemInfo = {
  id: "MLA931455240",
  title: "Apple iPhone 11 (128 Gb) - Blanco",
  category_id: "MLA1055",
  price: 301999,
  base_price: 301999,
  original_price: null,
  currency_id: "ARS",
  initial_quantity: 228,
  available_quantity: 1,
  sold_quantity: 200,
  condition: "new",
  permalink:
    "https://articulo.mercadolibre.com.ar/MLA-931455240-apple-iphone-11-128-gb-blanco-_JM",
  thumbnail_id: "796892-MLA46114829828_052021",
  thumbnail: "http://http2.mlstatic.com/D_796892-MLA46114829828_052021-I.jpg",
  secure_thumbnail:
    "https://http2.mlstatic.com/D_796892-MLA46114829828_052021-I.jpg",
  shipping: {
    free_shipping: true,
  },
};
