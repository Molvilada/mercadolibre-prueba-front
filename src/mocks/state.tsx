export const state = {
  categories: [
    {
      id: "MLA409810",
      name: "Audio",
      results: 282,
    },
    {
      id: "MLA1648",
      name: "Computación",
      results: 14,
    },
    {
      id: "MLA3502",
      name: "Accesorios para Celulares",
      results: 100,
    },
    {
      id: "MLA3813",
      name: "Repuestos de Celulares",
      results: 11,
    },
    {
      id: "MLA4641",
      name: "Pilas y Cargadores",
      results: 7,
    },
    {
      id: "MLA1055",
      name: "Celulares y Smartphones",
      results: 5,
    },
    {
      id: "MLA58775",
      name: "Fundas y Bolsos",
      results: 5,
    },
    {
      id: "MLA3690",
      name: "Accesorios para Audio y Video",
      results: 2,
    },
    {
      id: "MLA5054",
      name: "Cables",
      results: 2,
    },
    {
      id: "MLA11830",
      name: "Componentes Electrónicos",
      results: 2,
    },
    {
      id: "MLA4632",
      name: "Controles Remotos",
      results: 2,
    },
    {
      id: "MLA407134",
      name: "Herramientas",
      results: 2,
    },
    {
      id: "MLA5725",
      name: "Accesorios para Vehículos",
      results: 1,
    },
    {
      id: "MLA1039",
      name: "Cámaras y Accesorios",
      results: 1,
    },
    {
      id: "MLA1276",
      name: "Deportes y Fitness",
      results: 1,
    },
    {
      id: "MLA1499",
      name: "Industrias y Oficinas",
      results: 1,
    },
    {
      id: "MLA417704",
      name: "Smartwatches y Accesorios",
      results: 1,
    },
    {
      id: "MLA1915",
      name: "Otros",
      results: 4,
    },
    {
      id: "MLA1953",
      name: "Otras categorías",
      results: 1,
    },
  ],
  items: [
    {
      id: "MLA1274308983",
      title:
        "Cable Usb 2.0 Apple Lightning Blanco Con Entrada Usb Salida Lightning",
      price: {
        currency: "ARS",
        amount: 600,
        decimals: 0,
      },
      picture: "http://http2.mlstatic.com/D_918400-MLA52464082860_112022-I.jpg",
      condition: "new",
      free_shipping: false,
      address: "Buenos Aires",
    },
    {
      id: "MLA1230896865",
      title: "Apple AirPods (3ª generación) Con estuche De carga Lightning",
      price: {
        currency: "ARS",
        amount: 28116,
        decimals: 0,
      },
      picture: "http://http2.mlstatic.com/D_615592-MLM51650325727_092022-I.jpg",
      condition: "new",
      free_shipping: true,
      address: "Buenos Aires",
    },
    {
      id: "MLA1319294265",
      title: "Apple Earpods Con Conector De 3.5 Mm - Blanco",
      price: {
        currency: "ARS",
        amount: 4399,
        decimals: 0,
      },
      picture: "http://http2.mlstatic.com/D_831247-MLA46346545723_062021-I.jpg",
      condition: "new",
      free_shipping: false,
      address: "Buenos Aires",
    },
    {
      id: "MLA1144081798",
      title: "Funda Silicona Animada Para AirPods Inpods 12 I12 Diseños",
      price: {
        currency: "ARS",
        amount: 999,
        decimals: 0,
      },
      picture: "http://http2.mlstatic.com/D_824016-MLA50446569499_062022-I.jpg",
      condition: "new",
      free_shipping: false,
      address: "Buenos Aires",
    },
  ],
  bestCategory: [
    {
      id: "MLA409810",
      name: "Audio",
      results: 282,
    },
  ],
};
