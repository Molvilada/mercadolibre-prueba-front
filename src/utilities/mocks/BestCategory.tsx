export const bestCategory = [
  {
    id: "MLA1051",
    name: "Celulares y Tel√©fonos",
  },
  {
    id: "MLA1055",
    name: "Celulares y Smartphones",
  },
];
