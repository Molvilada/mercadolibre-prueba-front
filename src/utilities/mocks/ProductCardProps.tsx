export const productCardProps = {
  id: "MLA825206827",
  image: "http://http2.mlstatic.com/D_865864-MLA46114990464_052021-I.jpg",
  price: 301000,
  title: "Apple iPhone 11 (128 Gb) - Negro",
  address: "Capital Federal",
  last: false,
};
