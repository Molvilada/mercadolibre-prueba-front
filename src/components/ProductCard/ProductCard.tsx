import React from "react";
import Grid from "@mui/material/Grid";
import "./ProductCard.scss";
import { useNavigate } from "react-router-dom";
import { priceFormat } from "../../utilities/funcUtilities";
import shipping from "../../assets/images/ic_shipping.webp";

type Props = {
  id: string;
  image: string;
  price: number;
  title: string;
  address: string;
  last: boolean;
};

const ProductCard: React.FC<Props> = ({
  id,
  image,
  price,
  title,
  address,
  last,
}) => {
  const navigate = useNavigate();

  const handleClick = () => {
    navigate(`/items/${id}`);
  };
  return (
    <Grid container className="ProductCard" onClick={handleClick}>
      <Grid item xs="auto">
        <div className="productImg">
          <img src={image} alt={`${title}--img`}></img>
        </div>
      </Grid>
      <Grid item xs style={{ paddingTop: "1rem", paddingLeft: "1rem" }}>
        <div className="priceShipping">
          <h2>{priceFormat(price)}</h2>
          <img src={shipping} alt="shipping" />
        </div>
        <h3>{title}</h3>
      </Grid>
      <Grid
        item
        xs={2}
        style={{ paddingTop: "2rem", paddingLeft: "1rem", fontSize: "0.85rem" }}
        className="address"
      >
        {address}
      </Grid>
      {!last && (
        <Grid item xs={12}>
          <hr />
        </Grid>
      )}
    </Grid>
  );
};

export default ProductCard;
