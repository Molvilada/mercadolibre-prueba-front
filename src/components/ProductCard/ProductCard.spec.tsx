import React from "react";
import renderer from "react-test-renderer";
import { productCardProps } from "../../mocks/productCardProps";
import ProductCard from "./ProductCard";

describe("components/ProductCard", () => {
  test("ProductCard should render Correctly", () => {
    const container = renderer.create(<ProductCard {...productCardProps} />);
    expect(container.toJSON()).toMatchSnapshot();
  });
});
