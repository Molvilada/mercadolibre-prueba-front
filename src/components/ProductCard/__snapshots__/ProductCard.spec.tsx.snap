// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ProductCard ProductCard should render Correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container ProductCard css-11lq3yg-MuiGrid-root"
  onClick={[Function]}
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
  >
    <div
      className="productImg"
    >
      <img
        alt="Apple iPhone 11 (128 Gb) - Negro--img"
        src="http://http2.mlstatic.com/D_865864-MLA46114990464_052021-I.jpg"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
    style={
      Object {
        "paddingLeft": "1rem",
        "paddingTop": "1rem",
      }
    }
  >
    <div
      className="priceShipping"
    >
      <h2
        className="price"
      >
        $ 
        301.000
        <span
          className="decimals"
        >
          00
        </span>
      </h2>
      <img
        alt="shipping"
        src="ic_shipping.webp"
      />
    </div>
    <h3
      className="title"
    >
      Apple iPhone 11 (128 Gb) - Negro
    </h3>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 address css-1o7apob-MuiGrid-root"
    style={
      Object {
        "fontSize": "0.75rem",
        "paddingLeft": "1rem",
        "paddingTop": "2rem",
      }
    }
  >
    Capital Federal
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <hr />
  </div>
</div>
`;
